src/sfizz/FilePool.h

```diff
@@ -69,18 +68,30 @@ struct FilePromise
         preloadedData.reset();
         filename = "";
         availableFrames = 0;
-        dataReady = false;
+        dataStatus = DataStatus::Wait;
         oversamplingFactor = config::defaultOversamplingFactor;
         sampleRate = config::defaultSampleRate;
     }
 
+    void waitCompletion()
+    {
+        while (dataStatus == DataStatus::Wait)
+            std::this_thread::sleep_for(std::chrono::milliseconds(50));
+    }
+
+    enum class DataStatus {
+        Wait = 0,
+        Ready,
+        Error,
+    };
+
     absl::string_view filename {};
     AudioBufferPtr preloadedData {};
     AudioBuffer<float> fileData {};
     float sampleRate { config::defaultSampleRate };
     Oversampling oversamplingFactor { config::defaultOversamplingFactor };
     std::atomic<size_t> availableFrames { 0 };
-    std::atomic<bool> dataReady { false };
+    std::atomic<DataStatus> dataStatus { DataStatus::Wait };
```

Comment:
I added a check for lock-freeness, but other than this I think we can keep this enum state even if we don't use the Filepool to load wavetables.

Reply:
The purpose was to have a value here that will be set even in error case, such that I may know when to stop waiting if it failed.

Reply:
Yes it's good, no problem. The check is there to ensure no hidden mutex, although enum class probably has the same atomic properties as integers.

Reply:
Should this be changed to int to make sure ?

Reply:
Apparently no, from https://en.cppreference.com/w/cpp/language/enum

> 1) declares a scoped enumeration type whose underlying type is int (the keywords class and struct are exactly equivalent)

so if int is lock-free this should be.